official/utils/testing/golden_test.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""This module tests generic behavior of golden tests.
+
+  This test is not intended to test every layer of interest, and models should
+test the layers that affect them. This test is primarily focused on ensuring
+that golden.BaseTest functions as intended.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import sys
+import warnings
+
+import tensorflow as tf
+from official.utils.testing import golden  # pylint: disable=g-bad-import-order
```

Comment:
Reminder: Build files will be needed for all of this.
nit: add the bad import line to the tf import, and then if there are other official imports, you won't have to add the flag for all.

Reply:
Indeed. Also putting the bad-import on the tf line saved me a duplicate in layer_test.py. We certainly have our priorities in order!